src/node/core.go

```diff
@@ -417,6 +417,13 @@ func (c *Core) Leave(leaveTimeout time.Duration) error {
 		return nil
 	}
 
+	// Check for maintenance mode, if set no need for a leave request
+	if badgerStore, ok := c.hg.Store.(*hg.BadgerStore); ok {
+		if badgerStore.GetMaintenanceMode() {
```

Comment:
Here we could use the config.MaintenanceMode flag instead of this cast. That would require adding an extra parameter to the constructor of the core object, but it's probably worth it.